### PR TITLE
Copy button for ban/disconnect reason

### LINKED ISSUE #28560: Ban messages should be copyable
If your connection is denied due to a ban, there should be a button or something that copies the ban reason to your clipboard.

QOL for players making ban appeals since they're asked for the message and it'd make appeals more accurate since I see a lot of ban messages that aren't copied very well

### PR DESCRIPTION
**Pull request by Titian3**
## About the PR
Added copy button for disconnection and connection failure reasons on connection. Fixes #28560

## Why / Balance
When people are trying to relay connection failures to devs or ban reasons in appeal or to admins. They often have to write out the message or screencap to share it. Now we have modern copy pasta technology.

## Technical details
Add button to ui, use existing clipboard class to copy.

## Media
Examples/tests:
- Failure to connect:
image
Failed to connect to server:
Failed to establish connection - no response from remote host
----
- Disconnection:
image
`Kicked by console`
----
- Ban - timed:
image
`You, or another user of this computer or connection, are banned from playing here.
The ban reason is: "Self Antag - Forcing all crew to be a clown as acting captain."
This ban is for 20 minutes and will expire at Thursday, August 8, 2024 12:00 AM UTC.
Attempts to circumvent this ban such as creating a new account will be logged.`
----
- Ban - indef
image
`You, or another user of this computer or connection, are banned from playing here.
The ban reason is: "Self antag - murderboned all the crew as hamlet."
This ban will only be removed via appeal.
Attempts to circumvent this ban such as creating a new account will be logged.`
----

## Requirements
- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
None

**Changelog**
:cl: Repo
- add: Added copy to clipboard button for connection failure UI.

### CODE REVIEW

**Comment by metalgearsloth** [comment]
Why not just make the text highlightable?

**Comment by PJB3005** [comment]
> Why not just make the text highlightable?

Because we don't have that technology?

File: Content.Client/Launcher/LauncherConnectingGui.xaml.cs [hunk 1]
```diff
@@ -38,14 +39,24 @@ public sealed partial class LauncherConnectingGui : Control
...
 
             LayoutContainer.SetAnchorPreset(this, LayoutContainer.LayoutPreset.Wide);
 
             Stylesheet = IoCManager.Resolve<IStylesheetManager>().SheetSpace;
 
             ChangeLoginTip();
-            ReconnectButton.OnPressed += ReconnectButtonPressed;
-            RetryButton.OnPressed += ReconnectButtonPressed;
+            RetryButton.OnPressed += _ => _state.RetryConnect();
+            ReconnectButton.OnPressed += _ =>
```
Comment (PJB3005) [inline Content.Client/Launcher/LauncherConnectingGui.xaml.cs:50]: Why are you changing the code for the reconnect buttons?

File: Resources/Locale/en-US/launcher/launcher-connecting.ftl [hunk 2]
```diff
@@ -18,3 +18,4 @@ connecting-state-ResolvingHost = Resolving host
 connecting-state-EstablishingConnection = Establishing connection
 connecting-state-Handshake = Handshake
 connecting-state-Connected = Connected
+copy-reason = Copy Message
```
Comment (PJB3005) [inline Resources/Locale/en-US/launcher/launcher-connecting.ftl:21]: This localization key needs to start with `connecting-`  and also `-reason` doesn't make much sense.

**Review by PJB3005 — CHANGES_REQUESTED** [review]
Please use OpenLeft/OpenRight style classes so the buttons visually connect.

File: Content.Client/Launcher/LauncherConnectingGui.xaml.cs (hunk 1 above)
Comment (Titian3) [reply Content.Client/Launcher/LauncherConnectingGui.xaml.cs]: I think at the time which was late, I thought it was going to read better but looking at it now it is pain. Changed it back.

**Comment by Titian3** [comment]
> Please use OpenLeft/OpenRight style classes so the buttons visually connect.

image

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**